### ISSUE TITLE
[Windows] global CSS imports regression (blocks the Create Your First App learn step)

### ISSUE THREAD

**Issue by healeycodes** [vercel/next.js#36113]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

```
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Home
    Binaries:
      Node: 16.13.2
      npm: 7.24.0
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.5 (broken on 12.1.5-canary.7 as well)
      react: 17.0.2
      react-dom: 17.0.2
```

### What browser are you using? (if relevant)

Not relevant

### How are you deploying your application? (if relevant)

Not relevant

### Describe the Bug

I'm blocked on this step of the learn tutorial: https://nextjs.org/learn/basics/create-nextjs-app/setup.

Seems like a regression of this issue:
- https://github.com/vercel/next.js/issues/28579

Which was fixed in this PR:
- https://github.com/vercel/next.js/pull/28631

But the fix is no longer present:
- https://github.com/vercel/next.js/blob/630bf80a22132a08be11e001a4ab3ee0124dd5a2/packages/next/build/webpack-config.ts#L1749

<br>

User sees a _Failed to compile_ error in the browser, and the following in their terminal:

```
> dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 1957 ms (112 modules)
wait  - compiling / (client and server)...
error - ./pages/index.js
Module not found: Can't resolve 'styled-jsx/style' in 'C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\pages'
(node:24292) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
(Use `node --trace-deprecation ...` to show where the warning was created)
wait  - compiling /_error (client and server)...
error - ./pages/index.js
Module not found: Can't resolve 'styled-jsx/style' in 'C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\pages'
```

### Expected Behavior

`npx create-next-app nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"` should create a working application on Windows with latest `next`.

### To Reproduce

- `npx create-next-app nextjs-blog --use-npm --example "https://github.com/vercel/next-learn/tree/master/basics/learn-starter"`
- `cd nextjs-blog/`
- `npm run dev`

Output:

```
> dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 1957 ms (112 modules)
wait  - compiling / (client and server)...
error - ./pages/index.js
Module not found: Can't resolve 'styled-jsx/style' in 'C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\pages'
(node:24292) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
(Use `node --trace-deprecation ...` to show where the warning was created)
wait  - compiling /_error (client and server)...
error - ./pages/index.js
Module not found: Can't resolve 'styled-jsx/style' in 'C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\pages'
```

Output with `--trace-deprecation`:

```
NODE_OPTIONS='--trace-deprecation' npm run dev

> dev     
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 501 ms (112 modules)
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 86 ms (113 modules)
wait  - compiling / (client and server)...
error - ./pages/index.js
Module not found: Can't resolve 'styled-jsx/style' in 'C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\pages'
(node:24108) [DEP_WEBPACK_MODULE_ISSUER] DeprecationWarning: Module.issuer: Use new ModuleGraph API
    at Function.getModuleGraphForModule (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\compiled\webpack\bundle5.js:46500:10)
    at NormalModule.get issuer [as issuer] (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\compiled\webpack\bundle5.js:44262:22)
    at findEntryModule (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\dev\hot-reloader.js:108:16)
    at erroredPages (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\dev\hot-reloader.js:119:29)
    at HotReloader.getCompilationErrors (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\dev\hot-reloader.js:629:33) 
    at DevServer.getCompilationError (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\dev\next-dev-server.js:709:47) 
    at DevServer.findPageComponents (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\dev\next-dev-server.js:665:43)  
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async DevServer.renderErrorToResponse (C:\Users\Andrew\Documents\nextjs-windows-repro\nextjs-blog\node_modules\next\dist\server\base-server.js:1151:26)
```

**Comment by balazsorban44** [comment]
This issue has been automatically closed because it wasn't verified against next@canary. If you think it was closed by accident, please leave a comment. If you are running into a similar issue, please open a new issue with a reproduction. Thank you.
